File: December/Day16/Implement_Queue_using_Stacks.cpp
class MyQueue {
public:
   stack<int>s1,s2;
    MyQueue() {
        
    }
    
    void push(int x) {
     s1.push(x);   
    }
    
    int pop() {
      while(!s1.empty())
      {
          s2.push(s1.top());
          s1.pop();
      }
      int x = s2.top();
      s2.pop();
      while(!s2.empty())
      {
          s1.push(s2.top());
          s2.pop();
      }
    return x;
    }
    
    int peek() {
     while(!s1.empty())
      {
          s2.push(s1.top());
          s1.pop();
      }
      int x = s2.top();
      while(!s2.empty())
      {
          s1.push(s2.top());
          s2.pop();
      }
    return x;   
    }
    
    bool empty() {
      return s1.empty();  
    }
};

/**
 * Your MyQueue object will be instantiated and called as such:
 * MyQueue* obj = new MyQueue();
 * obj->push(x);
 * int param_2 = obj->pop();
 * int param_3 = obj->peek();
 * bool param_4 = obj->empty();
 */

File: December/Day16/implement_queue_using_stacks.py
class MyQueue:

    def __init__(self):
        self.stack1 = []
        self.stack2 = []

    def push(self, x: int) -> None:
        self.stack1.append(x)
        self.stack2.append(self.stack1.pop())

    def pop(self) -> int:
        return self.stack2.pop(0)

    def peek(self) -> int:
        return self.stack2[0]

    def empty(self) -> bool:
        if not self.stack2:
            return True
        return False
        